service/storage_service.cc

```diff
@@ -461,8 +461,13 @@ void storage_service::prepare_to_join(std::vector<inet_address> loaded_endpoints
         if (!is_auto_bootstrap()) {
             throw std::runtime_error("Trying to replace_address with auto_bootstrap disabled will not work, check your configuration");
         }
-        _bootstrap_tokens = prepare_replacement_info(loaded_peer_features).get0();
+
+        std::tie(_bootstrap_tokens, _streams_ts) = prepare_replacement_info(loaded_peer_features).get0();
+        // FIXME? (kbraun): I think we don't have to gossip tokens nor streams timestmap when HIBERNATEd
```

Comment:
The questions is: should the replacing node use the exact same cdc stream timestamp that the replaced was using?

Reply:
Hm, I made this before I introduced timestamp stealing (when a node sees someone is gossiping a higher timestamp, it will start gossiping it as well). Now it doesn't matter, I can as well remove _streams_ts from replacement info.

Reply:
Left it as it was: we use the replaced nodes' timestamp. It doesn't really matter: we have to use some timestamp, and we won't create a new generation of CDC streams, so we will use the replaced node's timestamp.

If it's not the newest one, the new node will soon switch to a newer timestamp on a gossip event anyway.

Maybe in the future we decide to not do this "gossiping the newest timestamp" for some reason, and then we won't have to change this replacement code.